test/integration/acceptance/transfer_asset_test.cpp

```diff
@@ -19,159 +19,103 @@ using namespace shared_model;
-   * @return pre-build tx
-   */
-  auto baseTx() {
-    return TestUnsignedTransactionBuilder()
-        .creatorAccountId(kUser1 + "@test")
-        .createdTime(getUniqueTime())
-        .quorum(1);
+  proto::Transaction makeTransfer(const std::string &amount) {
+    return complete(
+        baseTx().transferAsset(kUserId, kUser2Id, kAsset, kDesc, amount));
   }
 
-  /**
-   * Completes pre-build transaction
-   * @param builder is a pre-built tx
-   * @return built tx
-   */
-  template <typename TestTransactionBuilder>
-  auto completeTx(TestTransactionBuilder builder) {
-    return builder.build().signAndAddSignature(kUser1Keypair).finish();
+  proto::Transaction makeTransfer() {
+    return makeTransfer(kAmount);
   }
 
   const std::string kAmount = "1.0";
   const std::string kDesc = "description";
-  const std::string kUser1 = "userone";
-  const std::string kUser2 = "usertwo";
-  const std::string kRole1 = "roleone";
   const std::string kRole2 = "roletwo";
-  const std::string kUser1Id = kUser1 + "@test";
+  const std::string kUser2 = "usertwo";
   const std::string kUser2Id = kUser2 + "@test";
-  const crypto::Keypair kUser1Keypair =
-      crypto::DefaultCryptoAlgorithmType::generateKeypair();
   const crypto::Keypair kUser2Keypair =
       crypto::DefaultCryptoAlgorithmType::generateKeypair();
-  const interface::RolePermissionSet kPerms{
-      interface::permissions::Role::kAddAssetQty,
-      interface::permissions::Role::kTransfer,
-      interface::permissions::Role::kReceive};
 };
 
+#define check(i) [](auto &block) { ASSERT_EQ(block->transactions().size(), i); }
+
 /**
- * @given some user with all required permissions
+ * @given pair of users with all required permissions
  * @when execute tx with TransferAsset command
  * @then there is the tx in proposal
  */
 TEST_F(TransferAsset, Basic) {
   IntegrationTestFramework(1)
       .setInitialState(kAdminKeypair)
-      .sendTx(makeUserWithPerms(kUser1, kUser1Keypair, kPerms, kRole1))
-      .skipProposal()
-      .skipBlock()
-      .sendTx(makeUserWithPerms(kUser2, kUser2Keypair, kPerms, kRole2))
-      .skipProposal()
-      .skipBlock()
-      .sendTx(addAssets(kUser1, kUser1Keypair))
-      .skipProposal()
-      .skipBlock()
-      .sendTx(completeTx(
-          baseTx().transferAsset(kUser1Id, kUser2Id, kAsset, kDesc, kAmount)))
-      .checkBlock(
-          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .sendTxAwait(makeFirstUser(), check(1))
+      .sendTxAwait(makeSecondUser(), check(1))
+      .sendTxAwait(addAssets(), check(1))
+      .sendTxAwait(makeTransfer(), check(1))
       .done();
 }
 
 /**
- * @given some user with only can_transfer permission
+ * @given pair of users
+ *        AND the first user without can_transfer permission
  * @when execute tx with TransferAsset command
  * @then there is an empty proposal
  */
-TEST_F(TransferAsset, WithOnlyCanTransferPerm) {
+TEST_F(TransferAsset, DISABLED_WithOnlyCanTransferPerm) {
```

Comment:
Please specify in a comment the reason why the test is disabled. If the test can be enabled only after resolution of some other issue, denote it as a todo comment with Jira task id.